src/polynomial/univariate.rs

```diff
@@ -1,11 +1,16 @@
-use std::iter;
+use std::{
+    iter,
+    ops::{Add, Mul},
+};
 
-use crate::ff::Field;
+use halo2_proofs::halo2curves::ff::{PrimeField, WithSmallOrderMulGroup};
+
+use crate::{ff::Field, fft};
 
 /// Represents a univariate polynomial
 ///
 /// Coefficients of the polynomial are presented from smaller to larger
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Eq, Clone)]
```

Comment:
/// Coefficients of the polynomial are presented from smaller degree to larger degree